src/Type/MixedType.php

```diff
@@ -314,6 +314,10 @@ public function equals(Type $type): bool
 			return false;
 		}
 
+		if ($type instanceof ErrorType) {
```

Comment:
MixedType isn't equal to TemplateMixedType.

A bunch of types use `!$type instanceof static`, maybe we should do it in more cases.

Reply:
changing `MixedType->equals()` to return false for `TemplateMixedType` makes [this test fail](https://github.com/phpstan/phpstan-src/blob/cd2ca6483689a9805b8706c22897ea5413729b6d/tests/PHPStan/Generics/TemplateTypeFactoryTest.php#L50-L58).

I am not into generics enough to oversee how this should work

Reply:
ok, hopefully the new fix is what you had in mind